apis/v1/gateway_types.go

```diff
@@ -624,6 +644,31 @@ const (
 	TLSModePassthrough TLSModeType = "Passthrough"
 )
 
+// TLSConfig describes a TLS configuration defined per port.
+type TLSConfig struct {
```

Comment:
Recommend something like this to make it clear that this is per-port.
```suggestion
type TLSPortConfig struct {
```